endpoints/openrtb2/auction_test.go

```diff
@@ -120,196 +191,178 @@ func TestExplicitUserId(t *testing.T) {
+	assert.NoError(t, err, "Error jsonparsing root.code from file %s. Desc: %v.", testFile, err)
+
+	parsedTestData.ExpectedReturnCode = int(parsedReturnCode)
+
+	if parsedTestData.ExpectedReturnCode != 200 {
+		// Get expected error, fail if parsing error
+		parsedTestData.ExpectedErrorMessage, err = jsonparser.GetString(fileData, "expectedErrorMessage")
+		assert.NoError(t, err, "Error jsonparsing root.expectedErrorMessage from file %s. Desc: %v.", testFile, err)
+	} else {
+		// Get expected response, fail if parsing error
+		parsedTestData.ExpectedBidResponse, _, _, err = jsonparser.Get(fileData, "expectedBidResponse")
+		assert.NoError(t, err, "Error jsonparsing root.expectedBidResponse from file %s. Desc: %v.", testFile, err)
 	}
-	tests.assert(t)
-}
 
-// TestDisabledBidders makes sure we don't break when encountering a disabled bidder
-func TestDisabledBidders(t *testing.T) {
-	badTests := &getResponseFromDirectory{
-		dir:             "sample-requests/disabled/bad",
-		payloadGetter:   getRequestPayload,
-		messageGetter:   getMessage,
-		expectedCode:    http.StatusBadRequest,
-		aliased:         false,
-		disabledBidders: []string{"appnexus", "rubicon"},
-		adaptersConfig: map[string]config.Adapter{
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+	// Get testCaseConfig values
+	parsedTestData.Config = &testConfigValues{}
+	accReq, err := jsonparser.GetBoolean(fileData, "mockConfig", "accountRequired")
+	if err == nil {
+		parsedTestData.Config.AccountReq = accReq
 	}
-	goodTests := &getResponseFromDirectory{
-		dir:             "sample-requests/disabled/good",
-		payloadGetter:   noop,
-		messageGetter:   nilReturner,
-		expectedCode:    http.StatusOK,
-		aliased:         false,
-		disabledBidders: []string{"appnexus", "rubicon"},
-		adaptersConfig: map[string]config.Adapter{
-			"appnexus": {Disabled: true},
-			"rubicon":  {Disabled: true},
-		},
+
+	aliases, err := jsonparser.GetString(fileData, "mockConfig", "aliases")
+	if err == nil {
+		parsedTestData.Config.AliasJSON = aliases
 	}
-	badTests.assert(t)
-	goodTests.assert(t)
+
+	parsedTestData.Config.BlacklistedAccountArr = parseStringArray(fileData, "blacklistedAccts")
+	parsedTestData.Config.BlacklistedAppArr = parseStringArray(fileData, "blacklistedApps")
+	parsedTestData.Config.AdapterList = parseStringArray(fileData, "disabledAdapters")
+
+	return parsedTestData
 }
 
-// TestBlacklistRequests makes sure we return 400s on blacklisted requests.
-func TestBlacklistRequests(t *testing.T) {
-	// Need to turn off aliases for bad requests as applying the alias can fail on a bad request before the expected error is reached.
-	tests := &getResponseFromDirectory{
-		dir:           "sample-requests/blacklisted",
-		payloadGetter: getRequestPayload,
-		messageGetter: getMessage,
-		expectedCode:  http.StatusServiceUnavailable,
-		aliased:       false,
-	}
-	tests.assert(t)
+func parseStringArray(fileData []byte, jsonField string) []string {
+	rarr := []string{}
```

Comment:
`rarr` is a very confusing name. Consider calling it `results` or replace with using the standard library.

Reply:
`results` sounds great. Modified.